From b0c4628d221e8120d96cd9eba180e72880690882 Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Thu, 28 Aug 2014 14:22:45 -0400
Subject: [cpufreq] remove CONFIG_CPU_FREQ_TABLE

Message-id: <1409235879-1283-72-git-send-email-prarit@redhat.com>
Patchwork-id: 89391
O-Subject: [RHEL7.1 PATCH BZ 1134639 071/185] cpufreq: remove CONFIG_CPU_FREQ_TABLE
Bugzilla: 1134369
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Mark Langsdorf <mlangsdo@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1134639

commit 3bc28ab6da039f8020bbcea8e832b63a900bdb66
Author: Viresh Kumar <viresh.kumar@linaro.org>
Date:   Thu Oct 3 20:29:08 2013 +0530

    cpufreq: remove CONFIG_CPU_FREQ_TABLE

    CONFIG_CPU_FREQ_TABLE will be always enabled when cpufreq framework is used, as
    cpufreq core depends on it. So, we don't need this CONFIG option anymore as it
    is not configurable. Remove CONFIG_CPU_FREQ_TABLE and update its users.

    Signed-off-by: Viresh Kumar <viresh.kumar@linaro.org>
    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>

Cc: Lenny Szubowicz <lszubowi@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/cpufreq/Kconfig b/drivers/cpufreq/Kconfig
index 2d06754..38093e2 100644
--- a/drivers/cpufreq/Kconfig
+++ b/drivers/cpufreq/Kconfig
@@ -2,7 +2,6 @@ menu "CPU Frequency scaling"
 
 config CPU_FREQ
  bool "CPU Frequency scaling"
- select CPU_FREQ_TABLE
  help
    CPU Frequency scaling allows you to change the clock speed of 
    CPUs on the fly. This is a nice method to save power, because 
@@ -18,15 +17,11 @@ config CPU_FREQ
 
 if CPU_FREQ
 
-config CPU_FREQ_TABLE
- tristate
-
 config CPU_FREQ_GOV_COMMON
  bool
 
 config CPU_FREQ_STAT
  tristate "CPU frequency translation statistics"
- select CPU_FREQ_TABLE
  default y
  help
    This driver exports CPU frequency statistics information through sysfs
@@ -144,7 +139,6 @@ config CPU_FREQ_GOV_USERSPACE
 
 config CPU_FREQ_GOV_ONDEMAND
  tristate "'ondemand' cpufreq policy governor"
- select CPU_FREQ_TABLE
  select CPU_FREQ_GOV_COMMON
  help
    'ondemand' - This driver adds a dynamic cpufreq policy governor.
@@ -188,7 +182,6 @@ config CPU_FREQ_GOV_CONSERVATIVE
 config GENERIC_CPUFREQ_CPU0
  tristate "Generic CPU0 cpufreq driver"
  depends on HAVE_CLK && REGULATOR && PM_OPP && OF
- select CPU_FREQ_TABLE
  help
    This adds a generic cpufreq driver for CPU0 frequency management.
    It supports both uniprocessor (UP) and symmetric multiprocessor (SMP)
@@ -224,7 +217,6 @@ depends on IA64
 
 config IA64_ACPI_CPUFREQ
  tristate "ACPI Processor P-States driver"
- select CPU_FREQ_TABLE
  depends on ACPI_PROCESSOR
  help
  This driver adds a CPUFreq driver which utilizes the ACPI
@@ -241,7 +233,6 @@ depends on MIPS
 
 config LOONGSON2_CPUFREQ
  tristate "Loongson2 CPUFreq Driver"
- select CPU_FREQ_TABLE
  help
    This option adds a CPUFreq driver for loongson processors which
    support software configurable cpu frequency.
@@ -263,7 +254,6 @@ menu "SPARC CPU frequency scaling drivers"
 depends on SPARC64
 config SPARC_US3_CPUFREQ
  tristate "UltraSPARC-III CPU Frequency driver"
- select CPU_FREQ_TABLE
  help
    This adds the CPUFreq driver for UltraSPARC-III processors.
 
@@ -273,7 +263,6 @@ config SPARC_US3_CPUFREQ
 
 config SPARC_US2E_CPUFREQ
  tristate "UltraSPARC-IIe CPU Frequency driver"
- select CPU_FREQ_TABLE
  help
    This adds the CPUFreq driver for UltraSPARC-IIe processors.
 
@@ -286,7 +275,6 @@ menu "SH CPU Frequency scaling"
 depends on SUPERH
 config SH_CPU_FREQ
  tristate "SuperH CPU Frequency driver"
- select CPU_FREQ_TABLE
  help
    This adds the cpufreq driver for SuperH. Any CPU that supports
    clock rate rounding through the clock framework can use this
diff --git a/drivers/cpufreq/Kconfig.powerpc b/drivers/cpufreq/Kconfig.powerpc
index a9fbb3b..75918cb 100644
--- a/drivers/cpufreq/Kconfig.powerpc
+++ b/drivers/cpufreq/Kconfig.powerpc
@@ -1,7 +1,6 @@
 config CPU_FREQ_CBE
  tristate "CBE frequency scaling"
  depends on CBE_RAS && PPC_CELL
- select CPU_FREQ_TABLE
  default m
  help
    This adds the cpufreq driver for Cell BE processors.
@@ -20,7 +19,6 @@ config CPU_FREQ_CBE_PMI
 config CPU_FREQ_MAPLE
  bool "Support for Maple 970FX Evaluation Board"
  depends on PPC_MAPLE
- select CPU_FREQ_TABLE
  help
    This adds support for frequency switching on Maple 970FX
    Evaluation Board and compatible boards (IBM JS2x blades).
@@ -28,7 +26,6 @@ config CPU_FREQ_MAPLE
 config CPU_FREQ_PMAC
  bool "Support for Apple PowerBooks"
  depends on ADB_PMU && PPC32
- select CPU_FREQ_TABLE
  help
    This adds support for frequency switching on Apple PowerBooks,
    this currently includes some models of iBook & Titanium
@@ -37,7 +34,6 @@ config CPU_FREQ_PMAC
 config CPU_FREQ_PMAC64
  bool "Support for some Apple G5s"
  depends on PPC_PMAC && PPC64
- select CPU_FREQ_TABLE
  help
    This adds support for frequency switching on Apple iMac G5,
    and some of the more recent desktop G5 machines as well.
@@ -45,7 +41,6 @@ config CPU_FREQ_PMAC64
 config PPC_PASEMI_CPUFREQ
  bool "Support for PA Semi PWRficient"
  depends on PPC_PASEMI
- select CPU_FREQ_TABLE
  default y
  help
    This adds the support for frequency switching on PA Semi
diff --git a/drivers/cpufreq/Kconfig.x86 b/drivers/cpufreq/Kconfig.x86
index e2b6eab..6897ad8 100644
--- a/drivers/cpufreq/Kconfig.x86
+++ b/drivers/cpufreq/Kconfig.x86
@@ -31,7 +31,6 @@ config X86_PCC_CPUFREQ
 
 config X86_ACPI_CPUFREQ
  tristate "ACPI Processor P-States driver"
- select CPU_FREQ_TABLE
  depends on ACPI_PROCESSOR
  help
    This driver adds a CPUFreq driver which utilizes the ACPI
@@ -60,7 +59,6 @@ config X86_ACPI_CPUFREQ_CPB
 
 config ELAN_CPUFREQ
  tristate "AMD Elan SC400 and SC410"
- select CPU_FREQ_TABLE
  depends on MELAN
  ---help---
    This adds the CPUFreq driver for AMD Elan SC400 and SC410
@@ -76,7 +74,6 @@ config ELAN_CPUFREQ
 
 config SC520_CPUFREQ
  tristate "AMD Elan SC520"
- select CPU_FREQ_TABLE
  depends on MELAN
  ---help---
    This adds the CPUFreq driver for AMD Elan SC520 processor.
@@ -88,7 +85,6 @@ config SC520_CPUFREQ
 
 config X86_POWERNOW_K6
  tristate "AMD Mobile K6-2/K6-3 PowerNow!"
- select CPU_FREQ_TABLE
  depends on X86_32
  help
    This adds the CPUFreq driver for mobile AMD K6-2+ and mobile
@@ -100,7 +96,6 @@ config X86_POWERNOW_K6
 
 config X86_POWERNOW_K7
  tristate "AMD Mobile Athlon/Duron PowerNow!"
- select CPU_FREQ_TABLE
  depends on X86_32
  help
    This adds the CPUFreq driver for mobile AMD K7 mobile processors.
@@ -118,7 +113,6 @@ config X86_POWERNOW_K7_ACPI
 
 config X86_POWERNOW_K8
  tristate "AMD Opteron/Athlon64 PowerNow!"
- select CPU_FREQ_TABLE
  depends on ACPI && ACPI_PROCESSOR && X86_ACPI_CPUFREQ
  help
    This adds the CPUFreq driver for K8/early Opteron/Athlon64 processors.
@@ -132,7 +126,6 @@ config X86_POWERNOW_K8
 config X86_AMD_FREQ_SENSITIVITY
  tristate "AMD frequency sensitivity feedback powersave bias"
  depends on CPU_FREQ_GOV_ONDEMAND && X86_ACPI_CPUFREQ && CPU_SUP_AMD
- select CPU_FREQ_TABLE
  help
    This adds AMD-specific powersave bias function to the ondemand
    governor, which allows it to make more power-conscious frequency
@@ -160,7 +153,6 @@ config X86_GX_SUSPMOD
 
 config X86_SPEEDSTEP_CENTRINO
  tristate "Intel Enhanced SpeedStep (deprecated)"
- select CPU_FREQ_TABLE
  select X86_SPEEDSTEP_CENTRINO_TABLE if X86_32
  depends on X86_32 || (X86_64 && ACPI_PROCESSOR)
  help
@@ -190,7 +182,6 @@ config X86_SPEEDSTEP_CENTRINO_TABLE
 
 config X86_SPEEDSTEP_ICH
  tristate "Intel Speedstep on ICH-M chipsets (ioport interface)"
- select CPU_FREQ_TABLE
  depends on X86_32
  help
    This adds the CPUFreq driver for certain mobile Intel Pentium III
@@ -204,7 +195,6 @@ config X86_SPEEDSTEP_ICH
 
 config X86_SPEEDSTEP_SMI
  tristate "Intel SpeedStep on 440BX/ZX/MX chipsets (SMI interface)"
- select CPU_FREQ_TABLE
  depends on X86_32
  help
    This adds the CPUFreq driver for certain mobile Intel Pentium III
@@ -217,7 +207,6 @@ config X86_SPEEDSTEP_SMI
 
 config X86_P4_CLOCKMOD
  tristate "Intel Pentium 4 clock modulation"
- select CPU_FREQ_TABLE
  help
    This adds the CPUFreq driver for Intel Pentium 4 / XEON
    processors.  When enabled it will lower CPU temperature by skipping
@@ -259,7 +248,6 @@ config X86_LONGRUN
 
 config X86_LONGHAUL
  tristate "VIA Cyrix III Longhaul"
- select CPU_FREQ_TABLE
  depends on X86_32 && ACPI_PROCESSOR
  help
    This adds the CPUFreq driver for VIA Samuel/CyrixIII,
@@ -272,7 +260,6 @@ config X86_LONGHAUL
 
 config X86_E_POWERSAVER
  tristate "VIA C7 Enhanced PowerSaver (DANGEROUS)"
- select CPU_FREQ_TABLE
  depends on X86_32 && ACPI_PROCESSOR
  help
    This adds the CPUFreq driver for VIA C7 processors.  However, this driver
diff --git a/drivers/cpufreq/Makefile b/drivers/cpufreq/Makefile
index 7faffb3..e0cb177 100644
--- a/drivers/cpufreq/Makefile
+++ b/drivers/cpufreq/Makefile
@@ -1,5 +1,5 @@
 # CPUfreq core
-obj-$(CONFIG_CPU_FREQ)   += cpufreq.o
+obj-$(CONFIG_CPU_FREQ)   += cpufreq.o freq_table.o
 # CPUfreq stats
 obj-$(CONFIG_CPU_FREQ_STAT)             += cpufreq_stats.o
 
@@ -11,9 +11,6 @@ obj-$(CONFIG_CPU_FREQ_GOV_ONDEMAND) += cpufreq_ondemand.o
 obj-$(CONFIG_CPU_FREQ_GOV_CONSERVATIVE) += cpufreq_conservative.o
 obj-$(CONFIG_CPU_FREQ_GOV_COMMON)  += cpufreq_governor.o
 
-# CPUfreq cross-arch helpers
-obj-$(CONFIG_CPU_FREQ_TABLE)  += freq_table.o
-
 obj-$(CONFIG_GENERIC_CPUFREQ_CPU0) += cpufreq-cpu0.o
 
 ##################################################################################
diff --git a/drivers/cpufreq/cpufreq.c b/drivers/cpufreq/cpufreq.c
index f595b0a..cedc3e4 100644
--- a/drivers/cpufreq/cpufreq.c
+++ b/drivers/cpufreq/cpufreq.c
@@ -986,9 +986,7 @@ static void update_policy_cpu(struct cpufreq_policy *policy, unsigned int cpu)
 
  up_write(&per_cpu(cpu_policy_rwsem, policy->last_cpu));
 
-#ifdef CONFIG_CPU_FREQ_TABLE
  cpufreq_frequency_table_update_policy_cpu(policy);
-#endif
  blocking_notifier_call_chain(&cpufreq_policy_notifier_list,
    CPUFREQ_UPDATE_POLICY_CPU, policy);
 }
diff --git a/drivers/thermal/Kconfig b/drivers/thermal/Kconfig
index 5e3c025..21faa8a 100644
--- a/drivers/thermal/Kconfig
+++ b/drivers/thermal/Kconfig
@@ -69,7 +69,6 @@ config THERMAL_GOV_USER_SPACE
 config CPU_THERMAL
  bool "generic cpu cooling support"
  depends on CPU_FREQ
- select CPU_FREQ_TABLE
  help
    This implements the generic cpu cooling mechanism through frequency
    reduction. An ACPI version of this already exists
-- 
1.7.1